pkg/image/admission/imagequalify/sort.go

```diff
@@ -0,0 +1,19 @@
+package imagequalify
+
+import "github.com/openshift/origin/pkg/image/admission/imagequalify/api"
+
+// ByPatternPriority sorts a slice by pattern depth, in ascending
+// order.
+type ByPatternPriority []api.ImageQualifyRule
+
+func (x ByPatternPriority) Len() int {
+	return len(x)
+}
+
+func (x ByPatternPriority) Less(i, j int) bool {
+	return x[i].Pattern > x[j].Pattern
```

Comment:
is this just sorting by lexigraphic order?

Reply:
Yes; we want deeper paths to be compared first, and entries with wildcards should bind lower. This means we attempt to mach patterns with digest and tags before entries with just repository, then image name?

Reply:
Actually, I just wrote the following in doc.go; make sense?

`The rules will be automatically sorted based in ascending lexicographic
order of their patterns. For patterns, ascending means that longer paths
precede shorter paths.`

Reply:
consider these rules:
```yaml
- pattern: a
  domain: foo.com
- pattern: a*
  domain: bar.com
```

with the current logic, `a*` would sort before `a`, right?

combined with `path.Match` logic, that would mean that the image `a` would be qualified to `bar.com/a`, even though there was an exact match with the `a` rule

Reply:
I split the sorting of rules into two; first explicit rules, then wildcard rules. I also defined the order of sorting to be by pattern {depth, digest, tag, path}.  Wildcard rules will now always be matched after explicit rules.